zanata-client-commands/src/main/java/org/zanata/client/commands/push/RawPushCommand.java

```diff
@@ -75,10 +96,174 @@ public RawPushCommand(PushOptions opts, RestClientFactory clientFactory) {
         client = getClientFactory().getFileResourceClient();
     }
 
+    /**
+     * Extract extensions in square bracket
+     * input pattern: [extension;extension]
+     */
+    private List<String> extractExtensions(String rawExtensions) {
+        if(StringUtils.isEmpty(rawExtensions)) {
+            return Collections.emptyList();
+        }
+
+        //remove square bracket in first and last character
+        rawExtensions = rawExtensions.substring(1, rawExtensions.length()-1);
+        return Arrays.asList(rawExtensions.split(";"));
+    }
+
+    /**
+     * Extract input string to type and extensions
+     * input pattern: fileType[extension;extension], [extension;extension]
+     *
+     * @param typeWithExtension
+     * @return result[0] - type, result[1] - extensions. e.g [txt,html]
+     */
+    private String[] extractTypeAndExtension(String typeWithExtension) {
```

Comment:
Yeah, I think just having something like the following. It is not the most efficient, but it won't make a measurable difference to performance:

```
private static final Pattern fileNameExtensionsPattern = Pattern.compile("(.*)?(\\[(.*?)\\])?");

private String extractType(String typeWithExtension) {
    // run regex and extract group 1
}

private String extractExtensions(String typeWithExtension) {
    // run regex and extract group 2
}
```